src/ansys/simai/core/data/types.py

```diff
@@ -308,9 +309,12 @@ class RawSingleFilter(TypedDict):
 
 def to_raw_filters(filters: Optional["Filters"]) -> Optional["RawFilters"]:
     if isinstance(filters, dict):
-        return [{"field": k, "operator": "EQ", "value": v} for k, v in filters.items()]
+        tt = [json.dumps({"field": k, "operator": "EQ", "value": v}) for k, v in filters.items()]
+        return tt
```

Comment:
So it would return a `str[]` , thus breaking the return type of this function.

Function "to_raw_filters" is fine for me, it does it's job. In my opinion, it's in "iter_training_data" that you have to url_encode. ( maybe `query = urlencode([("filter[]", json.dumps(f)) for f in (filters or [])])` )

I'll go a bit further, but json.dumps produces an output that has spaces, so i'd rather add one parameter to dump :
```
json.dumps(data, separators=(',', ':'))
```

Reply:
yup, I implemented it too fast. I moved the encoding to `TrainingDataClientMixin.iter_training_data`, where the gets prepared, added the separators.